### DISCUSSION TITLE
official adapters

### DISCUSSION THREAD

**Discussion by jthegedus** [sveltejs/kit#2975] [General]
I may be mistaken in this, but from my understanding of the platforms supported by "official" adapters they have a 1:1 mapping of project/site to configurations. For example:

- 1 Netlify site == 1x `netlify.toml`
- 1 Vercel app == 1x `vercel.json`
etc

Monorepos, multiple-app repos or repos with other resources (APIs, DBs) are not supported by this model and require separate projects/sites to be created and mapped within the platforms (Netlify, Vercel). This 1:1 mapping isn't universal and I just want to raise that if the platforms of official adapters share this trait then it will implicitly be included during adapter API considerations.

As an example, Firebase supports multiple websites per project (X sites == 1x `firebase.json`) with the configuration for all living within a single configuration file. I'm not sure how many other platforms have this model, but I suspect more generic cloud platform adapters for say GCP/AWS/Azure may also fall under this category.

Again, just raising this trait which may not be apparent given the list of official adapters.

**Comment by benmccann** [comment]
Is there some way in which the API might need to change to support this consideration?

**Reply by jthegedus** [reply]
Not at this time, but I feel if I don't track all these adapter-related issues in this repo this particular trait may influence things without input from those building adapters outside of the model shared by the "official" adapters.

Related, I'm going to be afk for the better part of a month over the Dec/Jan holiday period and so won't be around to share inputs on this matter during the final push to 1.0. It would be a shame for this perspective to not be considered in the shaping of a 1.0 API that is then unable to be changed until 2.0.
